GitCommands/SshPathLocator.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.IO;
+using System.IO.Abstractions;
+using System.Linq;
+
+namespace GitCommands
+{
+    public interface ISshPathLocator
+    {
+        string Find(string gitBinDirectory);
+    }
+
+    public sealed class SshPathLocator : ISshPathLocator
+    {
+        private readonly IFileSystem _fileSystem;
+        private readonly IEnvironmentAbstraction _environment;
+
+        public SshPathLocator(IFileSystem fileSystem, IEnvironmentAbstraction environment)
+        {
+            _fileSystem = fileSystem;
+            _environment = environment;
+        }
+        public SshPathLocator()
+            : this(new FileSystem(), new EnvironmentAbstraction())
+        {
+        }
+
+
+        /// <summary>
+        /// Gets the git SSH command;
+        /// If the environment variable is not set, will try to find ssh.exe in git installation directory;
+        /// If not found, will return "".
+        /// </summary>
+        public string Find(string gitBinDirectory)
+        {
+            var ssh = _environment.GetEnvironmentVariable("GIT_SSH", EnvironmentVariableTarget.Process);
+
+            if (string.IsNullOrEmpty(ssh))
+            {
+                ssh = GetSshFromGitDir(gitBinDirectory);
+            }
+
+            return ssh ?? "";
+        }
+
+        private string GetSshFromGitDir(string gitBinDirectory)
+        {
+            if (string.IsNullOrEmpty(gitBinDirectory))
+            {
+                return null;
+            }
+            var gitDirInfo = _fileSystem.Directory.GetParent(gitBinDirectory);
```

Comment:
I would suggest to wrap filesystem calls into try..catch just in case. We don't want to crash the app just because we couldn't find ssh.exe.